my_each.rb

```diff
@@ -1,6 +1,168 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    array = to_a
+    (0...array.size).each do |i|
+      return to_enum(:my_each) unless block_given?
+
+      yield array[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    array = to_a
+    (0...array.size).each do |i|
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield array[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      return to_enum(:my_select) unless block_given?
+
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?(compare = nil)
+    result = true
+    my_each do |element|
+      if compare
+        result = compare.class.eql?(Class) ? element.is_a?(compare) : element.eql?(compare)
+      elsif block_given?
+        result = yield element
+
+      end
+      break unless result
+    end
+
+    result
+  end
+
+  def my_any?(compare = nil)
+    result = true
+    my_each do |element|
+      if compare
+        result = compare.class.eql?(Class) ? element.is_a?(compare) : element.eql?(compare)
+      elsif block_given?
+        result = yield element
+      end
+      break if result
+    end
+    result
+  end
+
+  def my_none?(compare = nil)
+    result = true
+    my_each do |element|
+      result = if compare
+                 compare.class.eql?(Class) ? !element.is_a?(compare) : !element.eql?(compare)
+               elsif block_given?
+                 !(yield element)
+               else
+                 !element
+               end
+      break unless result
+    end
+    result
+  end
```

Comment:
`my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/)   #should return true
%w[dog door rod blade].my_none?(/d/)   #should return false
```